rust/emitter/src/emitter.rs

```diff
@@ -707,6 +713,19 @@ impl InstructionWriter {
         self.emit_op(Opcode::JumpTarget);
     }
 
+    pub fn bytecode_offset(&mut self) -> BytecodeOffset {
+        BytecodeOffset { offset: self.bytecode.len() }
+    }
+
+    pub fn patch_jump_target(&mut self, jumplist: Vec<BytecodeOffset>) {
+        let target_offset = self.bytecode.len();
+        for jump in jumplist {
+            let index = jump.offset;
+            let new_target = (target_offset - index) as i32;
+            LittleEndian::write_i32(&mut self.bytecode[index..index + 4], new_target);
```

Comment:
`[index..index + 4]` includes the opcode.
this should be `[index+1..=index + 4]` so that this is the trailing 4 bytes after the opcode.

Reply:
Darn it. Sorry, Yulia.